### PR TITLE
[SofaBaseCollision] Clean code related to intersection methods

### PR DESCRIPTION
**Pull request by alxbilger**
- `DiscreteIntersection::testIntersection` and `DiscreteIntersection::computeIntersection` are made virtual, so derived classes can override them
- Comments
- Simplification
- Removed unused code (draw, unused variables etc)
- Fix `MinProximityIntersection::computeIntersection` and `NewProximityIntersection::computeIntersection` on cubes
- Fix compilation warnings

______________________________________________________

By submitting this pull request, I acknowledge that  
**I have read, understand, and agree [SOFA Developer Certificate of Origin (DCO)](https://github.com/sofa-framework/sofa/blob/master/CONTRIBUTING.md#sofa-developer-certificate-of-origin-dco)**.
______________________________________________________

**Reviewers will merge this pull-request only if**  
- it builds with SUCCESS for all platforms on the CI.
- it does not generate new warnings.
- it does not generate new unit test failures.
- it does not generate new scene test failures.
- it does not break API compatibility.
- it is more than 1 week old (or has fast-merge label).

### CODE REVIEW

**Comment by alxbilger** [comment]
[ci-build][with-all-tests]

File: SofaKernel/modules/SofaBaseCollision/src/SofaBaseCollision/MinProximityIntersection.cpp [hunk 1]
```diff
@@ -77,17 +76,12 @@ bool MinProximityIntersection::testIntersection(Cube& cube1, Cube& cube2)
...
-    return BaseProximityIntersection::testIntersection(cube1, cube2);
-}
-
-bool MinProximityIntersection::getUseSurfaceNormals(){
-    return useSurfaceNormals.getValue();
+    return BaseProximityIntersection::computeIntersection(cube1, cube2, contacts);
 }
 
-void MinProximityIntersection::draw(const core::visual::VisualParams* vparams)
+bool MinProximityIntersection::getUseSurfaceNormals() const
```
Comment (epernod) [inline SofaKernel/modules/SofaBaseCollision/src/SofaBaseCollision/MinProximityIntersection.cpp:82]: I find the name of the method strange. Maybe:
isUseSurfaceNormals  or getUseSurfaceNormalsValue()

File: SofaKernel/modules/SofaBaseCollision/src/SofaBaseCollision/MinProximityIntersection.h [hunk 2]
```diff
@@ -27,6 +27,29 @@
...
+ * - Line/Point (if useLinePoint is true)
+ * - Line/Line (if useLineLine is true)
+ * - Triangle/Point
+ * - Sphere/Point (if useSphereTriangle is true)
+ * - RigidSphere/Point (if useSphereTriangle is true)
+ * - Triangle/Sphere (if useSphereTriangle is true)
+ * - Triangle/RigidSphere (if useSphereTriangle is true)
+ * - Line/Sphere (if useSphereTriangle is true)
+ * - Line/RigidSphere (if useSphereTriangle is true)
+ * Note that MeshMinProximityIntersection ignores Triangle/Line and Triangle/Triangle intersections.
```
Comment (epernod) [inline SofaKernel/modules/SofaBaseCollision/src/SofaBaseCollision/MinProximityIntersection.h:50]: just for my knowledge, do you know why it is ignored?
The comment is very useful, thanks!

Comment (alxbilger) [reply SofaKernel/modules/SofaBaseCollision/src/SofaBaseCollision/MinProximityIntersection.h:50]: No I don't, but I know that if you want Triangle/Triangle intersection, you can go to `NewProximityIntersection`.

File: SofaKernel/modules/SofaBaseCollision/src/SofaBaseCollision/MinProximityIntersection.cpp (hunk 1 above)
Comment (alxbilger) [reply SofaKernel/modules/SofaBaseCollision/src/SofaBaseCollision/MinProximityIntersection.cpp:82]: GitHub formatting is not very clear, but I did not introduced `MinProximityIntersection::getUseSurfaceNormals()` myself. It was already in the code. And I agree with you @epernod

**Pull request merged**